Cancel interaction documentation - Fall 2019 documentation (#156)
smartdevicelink/sdl_app_library_guides: docs/Displaying a User Interface/Scrollable Message/index.md

```

!@

> ## Dismissing a Scrollable Message (RPC 6.0+)

Reviewer: ```suggestion
## Dismissing a Scrollable Message (RPC v6.0+)
```

```

!@

> ## Canceling a Specific Scrollable Message

Reviewer: ```suggestion
## Dismissing a Scrollable Message (RPC 6.0+)
```

> If you are connected to a head unit with SDL Core v6.0+, you can dismiss a displayed scrollable message before the timeout has elapsed.

Reviewer: ```suggestion
You can dismiss a displayed scrollable message before the timeout has elapsed. You can dismiss a specific scrollable message, or you can dismiss the scrollable message that is currently displayed.
```

> If connected to older head units that do not support this feature, the cancel request will be ignored, and the scrollable message will persist on the screen until the timeout has elapsed or the user dismisses the message by selecting a button.

Reviewer: Make a note

Author: made

### Dismissing a Specific Scrollable Message

@![iOS]
##### Objective-C
```objc
// Assign a unique cancel id to the scrollable message
UInt32 cancelID = 45;

> scrollableMessage.cancelID = @(cancelID);

Reviewer: This is confusing, remove

> // Use the cancel id to dismiss the scrollable message

Reviewer: ```suggestion
// `cancelID` is the ID that you assigned when creating and sending the scrollable message
```

SDLCancelInteraction *cancelInteraction = [[SDLCancelInteraction alloc] initWithScrollableMessageCancelID:cancelID];
[self.sdlManager sendRequest:cancelInteraction withResponseHandler:^(__kindof SDLRPCRequest * _Nullable request, __kindof SDLRPCResponse * _Nullable response, NSError * _Nullable error) {
    if (![response.resultCode isEqualToEnum:SDLResultSuccess]) { return; }
    <#The scrollable message was canceled successfully#>
}];
```

##### Swift
```swift
// Assign a unique cancel id to the scrollable message
let cancelID: UInt32 = 45

> scrollableMessage.cancelID = cancelID as NSNumber

Reviewer: See above, please remove

Author: removed

> // Use the cancel id to dismiss the scrollable message

Reviewer: ```suggestion
// `cancelID` is the ID that you assigned when creating and sending the alert
```

let cancelInteraction = SDLCancelInteraction(scrollableMessageCancelID: cancelID)
sdlManager.send(request: cancelInteraction) { (request, response, error) in
    guard response?.resultCode == .success else { return }
    <#The scrollable message was canceled successfully#>
}
```
!@

@![android,javaSE,javaEE]
```java
// Assign a unique cancel id to the scrollable message
final Integer cancelID = 45;

> scrollableMessage.setCancelID(cancelID);

Reviewer: See above, please remove

Author: removed

> // Use the cancel id to dismiss the scrollable message

Reviewer: ```suggestion
// `cancelID` is the ID that you assigned when creating and sending the alert
```
